src/arrayMethodReduce.js

```diff
@@ -5,7 +5,21 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let prev = initialValue;
+    let startIndex = prev === this[0]
+      ? 1
+      : 0;
+
+    if (arguments.length < 2) {
+      prev = this[0];
```

Comment:
you already have a check for `prev`.

check the previous comments. if you have any questions, feel free to ask in `fe_chat`

Reply:
*